cvat/apps/dataset_manager/tests/test_rest_api_formats.py

```diff
@@ -182,12 +288,794 @@ def _generate_url_dump_tasks_annotations(self, task_id):
+                    self.assertEqual(osp.exists(file_zip_name), True)
+
+                    self._remove_annotations(url, self.admin)
+
+                    self._create_annotations(task, "CVAT for images 1.1 many jobs", "default")
+
+                    if dump_format_name == "CVAT for images 1.1" or dump_format_name == "CVAT for video 1.1":
+                        dump_format_name = "CVAT 1.1"
+                    url = self._generate_url_upload_tasks_annotations(task_id, dump_format_name)
+
+                    with open(file_zip_name, 'rb') as binary_file:
+                        self._upload_file(url, binary_file, self.admin)
+                    task_ann = TaskAnnotation(task_id)
+                    task_ann.init_from_db()
+                    task_ann_data = task_ann.data
+                    self.assertEqual(len(task_ann_data["shapes"]), len(task_ann_prev_data["shapes"]))
+
+    def test_api_v1_unit_test_on_normalize_shape_function(self):
+        # 3 points
+        TrackManager.normalize_shape({
+             "points": [1.5, 2.5],
+        })
+
+        # 4 points
+        TrackManager.normalize_shape({
+            "points": [1.5, 2.5, 0.5, 8.6, 9.6, 3.6, 2.8, 7.2],
+        })
+        # 1 point
+        TrackManager.normalize_shape({
+            "points": [1.5, 2.5],
+        })
+        # empty shape
+        with self.assertRaises(ValueError):
+            TrackManager.normalize_shape({
+                "points": [],
+            })
+        # invalid count of points
+        with self.assertRaises(ValueError):
+            TrackManager.normalize_shape({
+                "points": [1.5, 2.5, 7.5],
+            })
+        # negative points
+        TrackManager.normalize_shape({
+            "points": [-1.5, 2.5, -9.8, -4.6],
+        })
+        # integer
+        TrackManager.normalize_shape({
+            "points": [1, 2, 9, 4],
+        })
+
+    def test_api_v1_tasks_annotations_dump_and_upload_many_jobs_with_datumaro(self):
+        test_name = self._testMethodName
+        upload_format_name = "CVAT 1.1"
+
+        for include_images in (False, True):
+            for dump_format_name in ("CVAT for images 1.1", "CVAT for video 1.1"):
+                with self.subTest():
+                    # create task with annotations
+                    images = self._generate_task_images(13)
+                    task = self._create_task(tasks["many jobs"], images)
+                    self._create_annotations(task, f'{dump_format_name} many jobs', "default")
+
+                    task_id = task["id"]
+                    task_ann = TaskAnnotation(task_id)
+                    task_ann.init_from_db()
+                    task_data = TaskData(task_ann.ir_data, Task.objects.get(pk=task_id))
+                    extractor = CvatTaskDataExtractor(task_data, include_images=include_images)
+                    data_from_task_before_upload = Dataset.from_extractors(extractor)
```

Comment:
I suspect this code can be extracted to a function, because it repeats multiple times.

Reply:
Done.